src/server/utils.js

```diff
@@ -1,7 +1,27 @@
 module.exports = {
+  parseArgv,
   getPage,
 }
 
+function parseArgv(arr) {
+  arr = typeof arr === 'string' ? arr.trim().split(/\s+/) : arr
+  return (arr || process.argv.slice(2)).reduce((acc, arg) => {
+    let [k, ...v] = arg.split(`=`)
+    v = v.join(`=`)
+    acc[k] =
+      v === ``
+        ? true
+        : /^(true|false)$/.test(v)
+        ? v === `true`
+        : /[\d|.]+/.test(v)
```

Comment:
Is this a good RegExp pattern for a number?  The following values for _v_ would match '.', '1.2.3' and 'test3test'.
I suggest the following pattern might be better `/^\d+(.\d+)?$/`. I would also suggest, if the acceptable range for the number can be safely determined, use `{1, L}` instead of `+`, where L is the maximum number of digits to exepect.

Reply:
``` js
function parseArgv(arr) { // Parse command-line arguments.
  return (arr || process.argv.slice(2)).reduce((acc, arg) => {
    let [k, ...v] = arg.split(`=`)
    v = v.join(`=`) // Combine values with `=` into a string.
    acc[k] = v === `` // Set to true when there's no value.
      ? true
      : (
        /^(true|false)$/.test(v) // Convert specified true/false.
        ? v === `true`
        : (
          /[\d|.]+/.test(v)
          ? (isNaN(Number(v)) ? v : Number(v)) // Use original string if the conversion to a number fails.
          : v
        )
      )
    return acc
  }, {})
}

parseArgv([`a`, `b=1`, `1.2.3`])
// {a: true, b: 1, 1.2.3: true}
```

It really doesn't take into account too many edge cases. All it does is convert booleans, strings, numbers. Do you have better code?

Reply:
If `.2` needs to be interpreted as a number then maybe `/^\d*(.\d+)?$/` would be better.

Reply:
you are right. Either regexp() or number() can do the same thing. I will optimize them later, thanks.